app/src/main/java/ch/epfl/sweng/eventmanager/ui/event/interaction/EventCreateActivity.java

```diff
@@ -296,24 +296,25 @@ protected void onActivityResult(int requestCode, int resultCode, @Nullable Inten
         if (data == null){
             return;
         }
-        if (requestCode == PICK_IMAGE && resultCode == RESULT_OK && data != null && data.getData() != null) {
+        if (requestCode == PICK_IMAGE && resultCode == RESULT_OK && data.getData() != null) {
             Uri eventImageUri = data.getData();
             cropAndConvertImage(eventImageUri);
         }
         if (resultCode == RESULT_OK && requestCode == UCrop.REQUEST_CROP) {
-                eventImageSrc = UCrop.getOutput(data);
-                imageLoader.displayImage(this, eventImageSrc, eventImage);
-            } else if (resultCode == UCrop.RESULT_ERROR) {
-                final Throwable cropError = UCrop.getError(data);
-                Log.i(TAG, "Unable to crop image");
-                cropError.printStackTrace();
-            }
+            eventImageSrc = UCrop.getOutput(data);
+            imageLoader.displayImage(this, eventImageSrc, eventImage);
+        } else if (resultCode == UCrop.RESULT_ERROR) {
+            final Throwable cropError = UCrop.getError(data);
+            Log.i(TAG, "Unable to crop image");
+            assert cropError != null;
```

Comment:
This line is not useful since assertions are removed at runtime... Furthermore, we can be sure cropError will not be null here .

Reply:
Changed!